packages/flame_test/test/flame_async_test.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:flame/components.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  var instructions = 0;
+
+  tearDown(() {
+    assert(instructions == 9);
```

Comment:
Note: Only way I could think on verifying that all of the method ran.

Reply:
Don't you want to do the `instructions++` from within the `async` context?

Reply:
But they are on the async context, here we just verify with the right number of instructions ran after the test is finished.

Reply:
I commented on the wrong line accidentally, I meant on line 15 and the likes. Very unlikely, but if `game.ensureAdd` returns without actually running the code in there `instructions` would still be bumped, but if you did `instructions++` from within the async operation that is running you would know that it has actually run.

Reply:
I still not sure I understood, is game.ensureAdd never completes, instructions will never run

Reply:
If the test would run some fake async where it returns directly on the `await`, they would still run `instructions++` but not necessarily the asynchronous code. But yeah, it would be unlikely that it would happen all of a sudden, so not worth to care about.

Reply:
Ah, got it, but AFAIK just by being async, is enough for the issue to arise, Component.onLoad don't run real async code, but without my fix, this test would break.

Reply:
Yeah, merge eeet!